autogpt/commands/file_operations.py

```diff
@@ -144,8 +147,18 @@ def write_to_file(filename: str, text: str) -> str:
         directory = os.path.dirname(filename)
         if not os.path.exists(directory):
             os.makedirs(directory)
-        with open(filename, "w", encoding="utf-8") as f:
-            f.write(text)
+        
+        if format == "json":
+            with open(filename, "w", encoding="utf-8") as f:
+                json.dump(text, f)
+        elif format == "csv":
+            with open(filename, "w", encoding="utf-8", newline='') as f:
+                writer = csv.writer(f)
+                writer.writerows(text)
```

Comment:
This doesn't make a lot of sense since `text` is explicitly typed as `str`. I'd advise breaking this up into separate commands.